domain/pool/pool.go

```diff
@@ -32,22 +32,26 @@ type VirtualIP struct {
 
 // ResourcePool A collection of computing resources with optional quotas.
 type ResourcePool struct {
-	ID                string        // Unique identifier for resource pool, eg "default"
-	Realm             string        // The name of the realm where this pool resides
-	Description       string        // Description of the resource pool
-	VirtualIPs        []VirtualIP   // All virtual IPs associated with a pool
-	CoreLimit         int           // Number of cores on the host available to serviced
-	MemoryLimit       uint64        // A quota on the amount (bytes) of RAM in the pool, 0 = unlimited
-	CoreCapacity      int           // Number of cores available as a sum of all cores on all hosts in the pool
-	MemoryCapacity    uint64        // Amount (bytes) of RAM available as a sum of all memory on all hosts in the pool
-	MemoryCommitment  uint64        // Amount (bytes) of RAM committed to services
-	ConnectionTimeout time.Duration // Wait delay on service rescheduling when an outage is reported
+	ID                string      // Unique identifier for resource pool, eg "default"
+	Realm             string      // The name of the realm where this pool resides
+	Description       string      // Description of the resource pool
+	VirtualIPs        []VirtualIP // All virtual IPs associated with a pool
+	CoreLimit         int         // Number of cores on the host available to serviced
+	MemoryLimit       uint64      // A quota on the amount (bytes) of RAM in the pool, 0 = unlimited
+	CoreCapacity      int         // Number of cores available as a sum of all cores on all hosts in the pool
+	MemoryCapacity    uint64      // Amount (bytes) of RAM available as a sum of all memory on all hosts in the pool
+	MemoryCommitment  uint64      // Amount (bytes) of RAM committed to services
+	ConnectionTimeout int         // Wait delay on service rescheduling when an outage is reported (milliseconds)
 	CreatedAt         time.Time
 	UpdatedAt         time.Time
 	MonitoringProfile domain.MonitorProfile
 	datastore.VersionedEntity
 }
 
+func (p ResourcePool) GetConnectionTimeout() time.Duration {
+	return time.Duration(p.ConnectionTimeout) * time.Millisecond
```

Comment:
No, because time.Millisecond is of type time.Duration--I have to cast ConnectionTimeout to be of the same type.